File: 0053-maximum-subarray/0053-maximum-subarray.c
#include <limits.h>

int maxSubArray(int* nums, int numsSize) {
    if (nums == NULL || numsSize == 0) return 0;

    int curSum = nums[0];
    int maxSum = nums[0];

    for (int i = 1; i < numsSize; i++) {
        if (curSum > 0) {
            curSum += nums[i];
        } else {
            curSum = nums[i];
        }

        if (curSum > maxSum) {
            maxSum = curSum;
        }
    }

    return maxSum;
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur_sum, max_sum = float('-inf'), float('-inf')

        for i in nums:
            if cur_sum > 0:
                cur_sum += i
            else:
                cur_sum = i
            
            if cur_sum > max_sum:
                max_sum = cur_sum

        return max_sum